workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs

```diff
@@ -62,53 +63,46 @@ protected override void OnUpdate()
             for (var i = 0; i < addedEntitiesData.Length; i++)
             {
                 var entity = addedEntitiesData.Entities[i];
-                var spatialEntityId = addedEntitiesData.SpatialEntityIds[i].EntityId;
-
-                var gameObject = gameObjectCreator.OnEntityCreated(
-                    new SpatialOSEntity(entity, EntityManager));
+                var spatialEntityId = addedEntitiesData.SpatialEntityIds[i];
+                var gameObject = gameObjectCreator.OnEntityCreated(new SpatialOSEntity(entity, EntityManager));
 
                 if (gameObject == null)
                 {
                     continue;
                 }
 
-                LinkerSystem.Linker.LinkGameObjectToEntity(gameObject, entity, spatialEntityId, PostUpdateCommands, viewCommandBuffer);
-                entityGameObjectCache.Add(entity, gameObject);
+                linkerSystem.Linker.LinkGameObjectToEntity(gameObject, entity, spatialEntityId.EntityId, viewCommandBuffer);
+                PostUpdateCommands.AddComponent(entity, new HasGameObjectSystemState { EntityId = spatialEntityId });
```

Comment:
Found one problem here actually... if the gameobject is null I am afraid `addedEntitiesData` will never be empty in following frames as long as entities are alive, probably because 
```
             [ReadOnly] public ComponentDataArray<NewlyAddedSpatialOSEntity> NewlyCreatedEntities; 
```

is no longer used

Reply:
good spot! will add the `NewlyAddedSpatialOSEntity` back into the struct